### PR TITLE
AccessReport model

### PR DESCRIPTION
**Pull request by t00f**
This model is intended to allow the enforcer to report:
- SSH logs
- User access to a specific API Service

> Project https://github.com/orgs/aporeto-inc/projects/480

### CODE REVIEW

File: specs/accessreport.spec [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    type: string
+    exposed: true
+    example_value: xxx-xxx-xxx-xxx
+
+  - name: processingUnitNamespace
+    description: Namespace of the processing unit of the report.
+    type: string
+    exposed: true
+    example_value: /my/ns
+
```
Comment (dstiliadis) [inline specs/accessreport.spec]: I suggest that we have a validation here that if processingunitID is provided, then namespace must be also provided (and vice versa).

File: specs/accessreport.spec [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  - name: reason
+    description: |-
+      This field is only set if 'action' is set to 'Reject' and specifies the reason
+      for the rejection.
+    type: string
+    exposed: true
+
```
Comment (dstiliadis) [inline specs/accessreport.spec]: Do we want to make this an enum for better UX? We had this problem with the flows.

File: specs/accessreport.spec [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  - name: type
+    description: Type of the report.
+    type: string
+    exposed: true
+    allowed_choices:
+    - SSHLogin
+    - SSHLogout
+    - SudoEnter
+    - SudoExit
+    example_value: SSHLogin
```
Comment (dstiliadis) [inline specs/accessreport.spec]: required

File: specs/accessreport.spec [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    description: |-
+      This field is only set if 'action' is set to 'Reject' and specifies the reason
+      for the rejection.
+    type: string
+    exposed: true
+
+  - name: timestamp
+    description: Date of the report.
+    type: time
+    exposed: true
```
Comment (dstiliadis) [inline specs/accessreport.spec]: required true

File: specs/accessreport.spec (hunk 1 above)
Comment (t00f) [reply specs/accessreport.spec]: Yes will do in a separate PR as it impacts all other policies

File: specs/accessreport.spec (hunk 2 above)
Comment (t00f) [reply specs/accessreport.spec]: I am hoping we could have a clear naming convention with a specific code. Each code would have a clear definition.

As a random example, we could have 
``` Rx1004 - No policy defined```
where:
* Rx indicate a code of reason
* 1000 indicates it is coming from the enforcer
* 4 indicates the type of error (in this case: no policy defined)

File: specs/accessreport.spec (hunk 4 above)
Comment (t00f) [reply specs/accessreport.spec]: All other flows are like this... is it an issue ? @primalmotion

File: specs/accessreport.spec (hunk 3 above)
Comment (t00f) [reply specs/accessreport.spec]: I used to have it required, but I need to set a default value. @primalmotion asked me to remove it. Should I put it back ?

File: specs/accessreport.spec (hunk 4 above)
Comment (primalmotion) [reply specs/accessreport.spec]: it's ok. backend adds a timestamp is 0.

File: specs/accessreport.spec (hunk 3 above)
Comment (primalmotion) [reply specs/accessreport.spec]: it needs a default value or an example_value. so put it back with an example_value. not a default_value

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "accessreport",
    "component": "gaia",
    "pr-id": "341",
    "commit-sha": "b532945415c59cff78f3aae037e474711afabb30"
  },
  {
    "project": "accessreport",
    "component": "backend",
    "pr-id": "435",
    "commit-sha": "68b3cd92e964ef7734a5e424b1939b768b047271"
  },
  {
    "project": "accessreport",
    "component": "apoctl",
    "pr-id": "184",
    "commit-sha": "e39e35896eaf2ada8664625e9d5c001fe59c382e"
  },
  {
    "project": "accessreport",
    "component": "enforcerd",
    "pr-id": "1243",
    "commit-sha": "448b28c2b8488196627ecc4f589e8658962df7db"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "accessreport",
    "component": "gaia",
    "pr-id": "341",
    "commit-sha": "b532945415c59cff78f3aae037e474711afabb30"
  },
  {
    "project": "accessreport",
    "component": "backend",
    "pr-id": "435",
    "commit-sha": "68b3cd92e964ef7734a5e424b1939b768b047271"
  },
  {
    "project": "accessreport",
    "component": "apoctl",
    "pr-id": "184",
    "commit-sha": "e39e35896eaf2ada8664625e9d5c001fe59c382e"
  },
  {
    "project": "accessreport",
    "component": "enforcerd",
    "pr-id": "1243",
    "commit-sha": "6cc6512460ff8c6f0eee2df4084958ef5b8a5e06"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "accessreport",
    "component": "gaia",
    "pr-id": "341",
    "commit-sha": "b532945415c59cff78f3aae037e474711afabb30"
  },
  {
    "project": "accessreport",
    "component": "backend",
    "pr-id": "435",
    "commit-sha": "68b3cd92e964ef7734a5e424b1939b768b047271"
  },
  {
    "project": "accessreport",
    "component": "apoctl",
    "pr-id": "184",
    "commit-sha": "e39e35896eaf2ada8664625e9d5c001fe59c382e"
  },
  {
    "project": "accessreport",
    "component": "enforcerd",
    "pr-id": "1243",
    "commit-sha": "a9d03e321ba8cf95dc20abcff2f7ac133bb1549e"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "accessreport",
    "component": "gaia",
    "pr-id": "341",
    "commit-sha": "b532945415c59cff78f3aae037e474711afabb30"
  },
  {
    "project": "accessreport",
    "component": "backend",
    "pr-id": "435",
    "commit-sha": "68b3cd92e964ef7734a5e424b1939b768b047271"
  },
  {
    "project": "accessreport",
    "component": "apoctl",
    "pr-id": "184",
    "commit-sha": "e39e35896eaf2ada8664625e9d5c001fe59c382e"
  },
  {
    "project": "accessreport",
    "component": "enforcerd",
    "pr-id": "1243",
    "commit-sha": "151c660dbd0af995c28b08ba1132186d67db4be1"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "accessreport",
    "component": "gaia",
    "pr-id": "341",
    "commit-sha": "0989ad73713901781b5a365784da77fc8657fe8f"
  },
  {
    "project": "accessreport",
    "component": "backend",
    "pr-id": "435",
    "commit-sha": "68b3cd92e964ef7734a5e424b1939b768b047271"
  },
  {
    "project": "accessreport",
    "component": "apoctl",
    "pr-id": "184",
    "commit-sha": "e39e35896eaf2ada8664625e9d5c001fe59c382e"
  },
  {
    "project": "accessreport",
    "component": "enforcerd",
    "pr-id": "1243",
    "commit-sha": "7df970ebaff1eb0989c01b46470186c7579bf95e"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "accessreport",
    "component": "apoctl",
    "pr-id": "184",
    "commit-sha": "e39e35896eaf2ada8664625e9d5c001fe59c382e"
  },
  {
    "project": "accessreport",
    "component": "enforcerd",
    "pr-id": "1243",
    "commit-sha": "7df970ebaff1eb0989c01b46470186c7579bf95e"
  },
  {
    "project": "accessreport",
    "component": "gaia",
    "pr-id": "341",
    "commit-sha": "0989ad73713901781b5a365784da77fc8657fe8f"
  },
  {
    "project": "accessreport",
    "component": "backend",
    "pr-id": "435",
    "commit-sha": "25c3f5292d5c4b4b0b22bfc51d35be971c692b12"
  }
]
```

**Pull request merged**